### PR TITLE
fix nll loss bug

### PR DESCRIPTION
**Pull request by yewentao256**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation and Context
fix Cuda total weight calculation issue
fix camb nll loss bugs

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: diopi_test/python/conformance/diopi_configs.py [hunk 1]
```diff
@@ -3279,100 +3279,47 @@
...
-    #         args=[
-    #             {
-    #                 "ins": ['input'],
-    #                 "requires_grad": [True],
-    #                 # "shape": ((0,), (16, 0,), (5, 0, 5, 6, 0, 3), (4, 0, 8, 3)),
-    #                 "shape": ((16, 0,), (5, 0, 5, 6, 0, 3), (4, 0, 8, 3)),
-    #             },
-    #             {
-    #                 "ins": ['target'],
-    #                 # "shape": ((), (16,), (5, 5, 6, 0, 3), (4, 8, 3)),
```
Comment (yangbofun) [inline diopi_test/python/conformance/diopi_configs.py:3362]: 这里验一下这个测例，是否torch可行。

Comment (yewentao256) [reply diopi_test/python/conformance/diopi_configs.py:3362]: 会出现nan的case，调整了一下做了适配

File: impl/camb/functions/loss.cpp [hunk 2]
```diff
@@ -195,15 +198,46 @@ diopiError_t diopiNLLLossBackward(diopiContextHandle_t ctx, diopiTensorHandle_t
...
     }
 
     auto gradInputRealTensor = requiresTensor(ctx, {n, c}, inputTensor.dtype());
-
     auto totalWeightTensor = requiresTensor(ctx, {1}, weightTensor.dtype());
+
+// For cnnl version（1.20.0), there is a bug in kernel causing grad_input
+// all zero randomly. We pass a wrong totalweight here.
+// TODO(someone): remove this when cnnl fixes this bug
+#if (CNNL_MAJOR * 10000 + CNNL_MINOR * 100 + CNNL_PATCHLEVEL <= 20000)
```
Comment (yangbofun) [inline impl/camb/functions/loss.cpp:210]: 这里，先把下面正确的实现，注释掉。把这个宏判断删掉。这里有风险。

Comment (yewentao256) [reply impl/camb/functions/loss.cpp]: 已使用注释方式处理

**Review by LeungChiNan — APPROVED** [review]

**Review by yangbofun — APPROVED** [review]
